src/Auction.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.27;
+
+import './math/AuctionMath.sol';
+import './Lending.sol';
+import "forge-std/console.sol";
+
+struct DeltaAuctionState {
+    int256 deltaFutureBorrowAssets;
+    int256 deltaFutureCollateralAssets;
+    int256 deltaUserCollateralAssets;
+    int256 deltaUserBorrowAssets;
+    int256 deltaUserFutureRewardCollateralAssets;
+    int256 deltaUserFutureRewardBorrowAssets;
+    int256 deltaProtocolFutureRewardCollateralAssets;
+    int256 deltaProtocolFutureRewardBorrowAssets;
+}
+
+event AuctionExecuted(DeltaAuctionState deltaState);
```

Comment:
Mb it's better to select what exactly we want to have here and emit few separate fields